### PR TITLE
fix(whitespace-issue): add validation to create label field

### PR DESCRIPTION
**Pull request by karthikjeeyar**
This PR contains the following changes

    1. Validates the create label input field, if validation fails disables the create button.
    2. Does not allow white space characters at the beginning of the label name.

 What issue/task does this PR references?
  
   https://github.com/openshiftio/openshift.io/issues/3117

Test cases - Added

### CODE REVIEW

**Comment by joshuawilson** [comment]
looks like you have a failing test `should disable the create label button` failed

File: src/app/components_ngrx/label-selector/label-selector.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        HttpClientTestingModule,
+        SelectDropdownModule,
+        StoreModule.forRoot({})
+      ],
+      declarations: [
+        LabelSelectorComponent
+      ],
+      providers: [ LabelService, HttpClientService, Store]
+    })
+        .compileComponents()
```
Comment (sanbornsen) [inline src/app/components_ngrx/label-selector/label-selector.component.spec.ts]: Incorrect indentation.

File: src/app/components_ngrx/label-selector/label-selector.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          spyOn(store, 'dispatch').and.callThrough();
+          fixture = TestBed.createComponent(LabelSelectorComponent);
+          componentInstance = fixture.componentInstance;
+          fixture.detectChanges();
+
+        });
+}));
+
+it(`should create label selector component`, async(() => {
+    expect(componentInstance).toBeTruthy();
```
Comment (sanbornsen) [inline src/app/components_ngrx/label-selector/label-selector.component.spec.ts]: Incorrect indentation.

File: src/app/components_ngrx/label-selector/label-selector.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+}));
+
+it('should recognize a label name field', async(() => {
```
Comment (sanbornsen) [inline src/app/components_ngrx/label-selector/label-selector.component.spec.ts]: All the `it` blocks are on incorrect indentation.

File: src/app/components_ngrx/label-selector/label-selector.component.ts [hunk 4]
```diff
@@ -210,6 +210,9 @@ export class LabelSelectorComponent implements OnInit {
   }
 
   onAddLabelInput(val) {
-    this.createDisabled = val === '';
+    if (val.trim().length === 0 && val.length) {
+      this.labelnameInput.nativeElement.value = '';
+    }
+    this.createDisabled = val.trim() === '';
```
Comment (sanbornsen) [inline src/app/components_ngrx/label-selector/label-selector.component.ts]: `this.createDisabled = true` can go inside the true. It's repeating condition check here. 
You can have a else block for `this.createDisabled = false`

**Pull request merged**